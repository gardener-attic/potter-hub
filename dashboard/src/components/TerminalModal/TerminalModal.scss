.MessageDetails {
  display: flex;
  align-items: center;
  justify-content: center;
  white-space: pre-wrap;
}
