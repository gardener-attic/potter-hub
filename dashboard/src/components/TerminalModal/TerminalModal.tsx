import * as React from "react";
import Modal from "react-modal";

import "./TerminalModal.css";

interface ITerminalModalProps {
  modalIsOpen: boolean;
  closeModal: () => Promise<any>;
  title: string;
  message: string;
}

const TerminalModal: React.SFC<ITerminalModalProps> = props => {
  return (
    <div className="MessageDetails">
      <Modal
        isOpen={props.modalIsOpen}
        onRequestClose={props.closeModal}
        contentLabel="Modal"
        className="Terminal"
        style={{
          overlay: {
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
          },
          content: {
            maxWidth: "80%",
          },
        }}
      >
        <div className="Terminal__Top type-small">
          <div className="Terminal__Top__Buttons">
            <a>
              <span
                className="Terminal__Top__Button Terminal__Top__Button--red"
                onClick={props.closeModal}
              />
            </a>
          </div>
          <div className="Terminal__Top__Title">{props.title}</div>
        </div>
        <div className="Terminal__Tab">
          <pre className="Terminal__Code">
            <code>{props.message}</code>
          </pre>
        </div>
      </Modal>
    </div>
  );
};

export default TerminalModal;
