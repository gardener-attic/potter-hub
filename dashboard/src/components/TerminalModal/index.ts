import TerminalModal from "./TerminalModal";

export default TerminalModal;
