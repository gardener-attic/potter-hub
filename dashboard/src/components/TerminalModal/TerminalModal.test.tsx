import { shallow } from "enzyme";
import * as React from "react";

import TerminalModal from "./TerminalModal";

it("renders the message details modal", () => {
  const wrapper = shallow(
    <TerminalModal
      modalIsOpen={false}
      title={"This is a title"}
      message={"This is a message!"}
      closeModal={jest.fn()}
    />,
  );

  expect(wrapper.find(".Terminal__Top__Title").text()).toBe("This is a title");
  expect(wrapper.find(".Terminal__Code").text()).toBe("This is a message!");
  expect(wrapper).toMatchSnapshot();
});
