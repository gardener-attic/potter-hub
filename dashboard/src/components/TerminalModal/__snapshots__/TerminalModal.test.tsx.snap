// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the message details modal 1`] = `
<div
  className="MessageDetails"
>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="Terminal"
    closeTimeoutMS={0}
    contentLabel="Modal"
    isOpen={false}
    onRequestClose={[MockFunction]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "maxWidth": "80%",
        },
        "overlay": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        },
      }
    }
  >
    <div
      className="Terminal__Top type-small"
    >
      <div
        className="Terminal__Top__Buttons"
      >
        <a>
          <span
            className="Terminal__Top__Button Terminal__Top__Button--red"
            onClick={[MockFunction]}
          />
        </a>
      </div>
      <div
        className="Terminal__Top__Title"
      >
        This is a title
      </div>
    </div>
    <div
      className="Terminal__Tab"
    >
      <pre
        className="Terminal__Code"
      >
        <code>
          This is a message!
        </code>
      </pre>
    </div>
  </Modal>
</div>
`;
